tket/src/Circuit/PauliExpBoxes.cpp

```diff
@@ -363,4 +363,272 @@ Op_ptr PauliExpCommutingSetBox::from_json(const nlohmann::json &j) {
 
 REGISTER_OPFACTORY(PauliExpCommutingSetBox, PauliExpCommutingSetBox)
 
+TermSequenceBox::TermSequenceBox(
+    const std::vector<SymPauliTensor> &pauli_gadgets,
+    Transforms::PauliSynthStrat synth_strategy,
+    PauliPartitionStrat partition_strategy, GraphColourMethod graph_colouring,
+    CXConfigType cx_configuration)
+    : Box(OpType::TermSequenceBox),
+      pauli_gadgets_(pauli_gadgets),
+      synth_strategy_(synth_strategy),
+      partition_strategy_(partition_strategy),
+      graph_colouring_(graph_colouring),
+      cx_configuration_(cx_configuration) {
+  // check at least one gadget
+  if (pauli_gadgets.empty()) {
+    throw PauliExpBoxInvalidity(
+        "TermSequenceBox requires at least one Pauli string");
+  }
+  // check all gadgets have same Pauli string length
+  auto n_qubits = pauli_gadgets[0].size();
+  for (const auto &gadget : pauli_gadgets) {
+    if (gadget.size() != n_qubits) {
+      throw PauliExpBoxInvalidity(
+          "the Pauli strings within TermSequenceBox must all be the "
+          "same length - add Pauli.I to pad strings to required length.");
+    }
+  }
+  signature_ = op_signature_t(n_qubits, EdgeType::Quantum);
+}
+
+TermSequenceBox::TermSequenceBox(const TermSequenceBox &other)
+    : Box(other),
+      pauli_gadgets_(other.pauli_gadgets_),
+      synth_strategy_(other.synth_strategy_),
+      partition_strategy_(other.partition_strategy_),
+      graph_colouring_(other.graph_colouring_),
+      cx_configuration_(other.cx_configuration_) {}
+
+TermSequenceBox::TermSequenceBox() : TermSequenceBox({{{}, 0}}) {}
+
+bool TermSequenceBox::is_clifford() const {
+  return std::all_of(
+      pauli_gadgets_.begin(), pauli_gadgets_.end(),
+      [](const SymPauliTensor &pauli_exp) {
+        return equiv_0(4 * pauli_exp.coeff) || pauli_exp.string.empty();
+      });
+}
+
+SymSet TermSequenceBox::free_symbols() const {
+  std::vector<Expr> angles;
+  for (const auto &pauli_exp : pauli_gadgets_) {
+    angles.push_back(pauli_exp.coeff);
+  }
+  return expr_free_symbols(angles);
+}
+
+Op_ptr TermSequenceBox::dagger() const {
+  std::vector<SymPauliTensor> dagger_gadgets;
+  for (const auto &pauli_exp : pauli_gadgets_) {
+    dagger_gadgets.emplace_back(pauli_exp.string, -pauli_exp.coeff);
```

Comment:
I'm not sure - I guess it makes things neater, but it adds minor extra logic for copying, mutating, and adding + it doesn't look like we expose the `transpose` method when adding python bindings so it may only be used here. I am leaning towards it being ok just changing the coefficient here, but am ofc happy to make the small changes required if you think it makes more sense to have a new method (in a separate pr?)